### ISSUE TITLE
Enhance support for Application Properties

### ISSUE THREAD

**Issue by vijaynalawade** [TIBCOSoftware/flogo#226]
**I'm submitting a ...**  (check one with "x")
```
[] bug report => search github for a similar issue or PR before submitting
[x] feature request
[] support request
[] general question
```

**Current behavior (how does the issue manifest):**
Currently, there is no support for property type or grouping of properties.
**Expected behavior:**
Users should be able to set a type as well able to group set of properties together.
**Minimal steps to reproduce the problem (not required if feature enhancement):**

**What is the motivation / use case for changing the behavior?**
https://github.com/TIBCOSoftware/flogo/wiki/Application-Properties
**Please tell us about your environment (Operating system, docker version, browser & web ui version, etc):**

**Flogo version (CLI & contrib/lib. If unknown, leave empty or state unknown):** 0.X.X

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by retgits** [comment]
I like the idea! One small question though, in one of the code blocks you're referring to `$env[ENV_DB_URL]` while today that would be `$env.ENV_DB_URL`. Do we expect to change that too and if so, will it be backward compatible with the existing apps?

**Comment by vijaynalawade** [comment]
Thats proposed change for consistency with $property and $activity. Though supported for few releases, we will deprecate use of  $env.<NAME> in the future.

**Comment by ayh20** [comment]
are you talking about deprecating $env completely ?  or just the $env.ENV_VAR syntax ?

**Comment by mellistibco** [comment]
@ayh20 just changing the syntax. so we'll make the switch to `$env[ENV_DB_URL]` and support the syntax of `$env.ENV_DB_URL` and `$env[ENV_DB_URL]` for a period of time.

**Comment by mellistibco** [comment]
Resolved in TIBCOSoftware/flogo-lib#171
